src/intermine_nlp/util.clj

```diff
@@ -0,0 +1,51 @@
+(ns intermine-nlp.util
+  (:require [clojure.edn :as edn]
+            [clojure.java.io :as io]
+            [clojure.pprint :refer [pprint]]
+            [imcljs.fetch :as fetch]
+            [imcljs.path :as path]
+            [clojure.core.async :refer [<! go]]
+            [clojure.string :refer [join split]]
+            [clj-http.client :as http]
+            [clojure.string :as str]))
+
+(defn read-edn
+  "Read an edn file into a map"
+  [path]
+  (try (with-open [file (io/reader path)]
+           (-> file java.io.PushbackReader. edn/read))
+       (catch Exception e (printf "Couldn't open file '%s'." path))))
+
+(defn pseudo-possible-values-
+  "imcljs.fetch/possible-values seems to be broken, so here's an implementation
+  using only imcljs.fetch/rows. 'path' must point to a field, not a class."
+  [service path]
+  (let [split-path (split (path/adjust-path-to-last-class (:model service) path) #"\.")
+        query {:from (first split-path)
+               :select [(last split-path)]}
+        rows (try (fetch/rows service query)
+                  (catch java.lang.Exception e nil))]
+       (->> rows :body :results (map first) distinct)))
+
+(def psuedo-possible-values
+    "imcljs.fetch/possible-values seems to be broken, so here's an implementation
```

Comment:
This is totally beyond minimal required scope, but if you have the time to PR a fix to imcljs that would be neat.